### PR TITLE
Remove playback fade-in and add comments to `AudioServer`

### LINKED ISSUE #66452: Godot 4 - Audio clips have their beginning cut off
### Godot version

v4.0.beta1.official [20d667284]

### System information

Windows 10, Vulkan, NVidia RTX 2070 Super

### Issue description

In Godot 4, some audio clips sound incorrect in engine for me; it sounds like the start is being cut off. I've tried both .ogg and .wav files and it happens with both, I've tried tweaking the import settings on the sound to no effect, and I've tried an identical setup in Godot 3.4.2 and the issue does not occur.

Here's a video comparing the original sound file, the sound in Godot 4, and the sound in Godot 3. You can hear that in Godot 4 the sound file sounds less "sharp" at the beginning as it does in the original file and Godot 3. Both these projects were created with no changes made to any settings.

https://user-images.githubusercontent.com/3895119/192319165-d8c13f8f-4ce7-4a0c-a10f-9e494e10ac65.mp4

### Steps to reproduce

1. Create a new project in Godot 4
2. Import a short sound like the one I used (It's most noticeable on very short 8-bit style sounds)
3. Play through the preview or through an AudioStreamPlayer and compare with the original sound file

### Minimal reproduction project

Godot 4.0 beta project: 
[Soundtest.zip](https://github.com/godotengine/godot/files/9647648/Soundtest.zip)

Godot 3.4.2 project:
[Soundtest3.zip](https://github.com/godotengine/godot/files/9647650/Soundtest3.zip)

### PR DESCRIPTION
**Pull request by KeyboardDanni**
Based on #87064. Manually redid the changes on top of latest, with a few tweaks. Let me know if I missed anything.

Fixes #66452.

### CODE REVIEW

**Review by adamscott — APPROVED** [review]
This looks fine. Thanks for salvaging the PR, @KeyboardDanni !

File: servers/audio_server.cpp [hunk 1]
```diff
@@ -371,10 +371,13 @@ void AudioServer::_mix_step() {
 			bus->soloed = false;
 		}
 	}
+	// This is legacy code from 3.x that allows video players and other audio sources that do not implement AudioStreamPlayback to output audio.
 	for (CallbackItem *ci : mix_callback_list) {
 		ci->callback(ci->userdata);
 	}
 
+	// Main mixing loop for audio streams.
+	// The basic idea here is to copy the samples returned by the AudioStreamPlayback's mix function into the audio buffers, while always maintaining a lookahead buffer of size LOOKAHEAD_BUFFER_SIZE to allow fade-outs for sudden stoppages.
```
Comment (AThousandShips) [inline servers/audio_server.cpp:380]:
```suggestion
	// The basic idea here is to copy the samples returned by the AudioStreamPlayback's mix function into the audio buffers,
	// while always maintaining a lookahead buffer of size LOOKAHEAD_BUFFER_SIZE to allow fade-outs for sudden stoppages.
```

File: servers/audio_server.cpp [hunk 2]
```diff
@@ -385,22 +388,25 @@ void AudioServer::_mix_step() {
 			continue;
 		}
 
+		// If `fading_out` is true, we're in the process of fading out the stream playback.
+		// TODO: Currently this sets the volume of the stream to 0 which creates a linear interpolation between its previous volume and silence. A more punchy option for fading out could be to just use the lookahead buffer.
```
Comment (AThousandShips) [inline servers/audio_server.cpp:392]:
```suggestion
		// TODO: Currently this sets the volume of the stream to 0 which creates a linear interpolation between its previous volume and silence.
		// A more punchy option for fading out could be to just use the lookahead buffer.
```

File: servers/audio_server.cpp [hunk 3]
```diff
@@ -416,42 +422,51 @@ void AudioServer::_mix_step() {
...
 					continue;
 				}
+				// If the StringNames of the buses match, we've found the previous bus index. This indicates that this playback mixed to `prev_bus_details->bus[prev_bus_index]` in the previous mix step, which gives us a way to look up the playback's previous volume.
 				if (playback->prev_bus_details->bus[search_idx].hash() == bus_details.bus[idx].hash()) {
 					prev_bus_idx = search_idx;
+					break;
 				}
 			}
 
+			// It's now time to mix to the bus. We do this by going through each channel of the bus and mixing to it. The channels correspond to output channels of the audio device, e.g. stereo or 5.1. To reduce needless nesting, this is done with a helper method named `_mix_step_for_channel`.
```
Comment (AThousandShips) [inline servers/audio_server.cpp:458]:
```suggestion
			// It's now time to mix to the bus. We do this by going through each channel of the bus and mixing to it.
			// The channels correspond to output channels of the audio device, e.g. stereo or 5.1. To reduce needless nesting, this is done with a helper method named `_mix_step_for_channel`.
```

**Review by AThousandShips — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**